### ISSUE TITLE
No useful message during errors received from endpoint

### ISSUE THREAD

**Issue by alkiskal** [eddieantonio/node-sparql-client#1]
The problem is here (client.js)

```
 if (response.statusCode >= 300) {
    throw new Error("SPARQL query failed.");
}
```

The response (message and status code) should be included in the error

**Comment by eddieantonio** [comment]
Started development in [this branch](https://github.com/eddieantonio/node-sparql-client/compare/v0.4.1...eddieantonio:feature/errors); it's not ready yet. Apparently the error removes any added projects, so I might need to use an Error subclass to add the status code.

**Comment by alkiskal** [comment]
That was fast!

**Comment by eddieantonio** [comment]
Fixed in version v0.4.2. From the README:

> ## Errors
> 
> If an error occurs, such as when submitting a query with a syntax error,
> the first argument to the `execute()` will be an `Error` object and have
> the `.httpStatus` attribute with the associated HTTP status code.
> Usually this is `400` when there is a syntax error, or `500` when the
> server refuses to process the request (such as when a timeout occurs).
> This status code is defined by the particular SPARQL server used.
> 
> ``` javascript
> new SparqlCLient(endpoint).query(`
>     SELECT ?name
>     WHERE { ?x foaf:name ?name
>     ORDER BY ?name
>   `)
>   .execute(function (err, data) {
>     console.log(err.httpStatus);
>     // logs '400'
>     console.log(err);
>     // logs 'HTTP Error: 400 Bad Request'
>   });
> ```
> 
> This also works with promises:
> 
> ``` javascript
> new SparqlCLient(endpoint).query(`
>     SELECT ?name
>     WHERE { ?x foaf:name ?name
>     ORDER BY ?name
>   `)
>   .execute()
>   .then(function () {
>     // will never reach here!
>   })
>   .catch(function (err) {
>     console.log(err.httpStatus);
>     // logs '400'
>     console.log(err);
>     // logs 'HTTP Error: 400 Bad Request'
>   });
> ```
